libraries/js/text/regex.effekt

```diff
@@ -184,8 +193,20 @@ namespace internal {
   extern io def unsafeExec(reg: Regex, str: String): RegexMatch = js "regex$exec(${reg}, ${str})"
 
   extern pure def unsafeRegex(str: String, flags: String): Regex = js "new RegExp(${str}, ${flags})"
+
+  extern pure def nodeMajorVersion(): Int = js """process$version()"""
 }
 
+def regexIndicesSupported?(): Bool = internal::nodeMajorVersion() >= 16
+
+extern js """
+  function process$version() {
+    const v = process.version
+    const parts = v.split('.')
+    return parseInt(parts[0].replace('v', ''))
+  }
```

Comment:
As a tiny nitpick: according to StackOverflow, `process.versions.node` should have the version without `v` :)
https://stackoverflow.com/questions/6656324/check-for-current-node-version